tests/test_qnode.py

```diff
@@ -1421,22 +1445,22 @@ def qn2(x, y):
 
     def test_shots_update_with_device(self):
         """Test that _shots is updated when updating the QNode with a new device."""
-        dev1 = qml.device("default.qubit", wires=1, shots=100)
-        qn = qml.QNode(dummyfunc, dev1)
+        dev1 = qml.device("default.qubit", wires=1)
+        qn = qml.set_shots(qml.QNode(dummyfunc, dev1), shots=100)
         assert qn._shots == qml.measurements.Shots(100)
 
         # _shots should take precedence over device shots
         with pytest.warns(
             UserWarning, match="The device's shots value does not match the QNode's shots value."
         ):
-            dev2 = qml.device("default.qubit", wires=1, shots=200)
+            dev2 = qml.device("default.qubit", wires=1)
```

Comment:
Here I think we actually do want to check the behavour when shots are on  the device. So probably one `with pytest.warns` for creating the device, and then a second `with pytest.warns` for updating the qnode.

Reply:
https://github.com/PennyLaneAI/pennylane/pull/7979/commits/6dbe604f526cf22e095e5649985d411ce979c2ca